src/BatteryStateOfChargeService.h

```diff
@@ -45,5 +45,12 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
     // Convention is to have an _ after private members
     const double initialStateOfChargePercent_;
 
+    double AmpHoursUsed_;
+    double inCurrent_;
+    double newCurrent_=0;
+    double sumCurrent_=0;
+    bool firstRun_=true;
```

Comment:
Also, all initializations should be done in the `.cpp` file. It gives a central point for readers to know how variables are initialized.This issue concerns lines 50-52, these were listed as warnings when you compiled, for some newer versions it comes up as a compiler error instead of a warning. To fix this:

```c++
BatteryStateOfChargeService::BatteryStateOfChargeService(double initialStateOfChargePercent)
 : initialStateOfChargePercent_(initialStateOfChargePercent)
 , newCurrent (0)
{
...
}
```
is the syntax to use. 
```c++
BatteryStateOfChargeService::BatteryStateOfChargeService(double initialStateOfChargePercent)
 : initialStateOfChargePercent_(initialStateOfChargePercent)
 {
   newCurrent = 0;
   ...
  }
```
is also valid but the other way is superior because of performance reasons (if you are curious https://isocpp.org/wiki/faq/ctors#init-lists)